patches/net/minecraft/world/item/ItemStack.java.patch

```diff
@@ -9,6 +9,17 @@
      public static final Codec<Holder<Item>> ITEM_NON_AIR_CODEC = BuiltInRegistries.ITEM
          .holderByNameCodec()
          .validate(
+@@ -234,6 +_,10 @@
+         return !this.isEmpty() ? this.components.asPatch() : DataComponentPatch.EMPTY;
+     }
+ 
++    public boolean isComponentsPatchEmpty() {
++        return !this.isEmpty() ? this.components.isPatchEmpty() : true;
```

Comment:
Boolean ternaries are always harder to read, why not just write this as `this.isEmpty() || this.components.isPatchEmpty()`?

Reply:
Just because it matches the vanilla functions above. :P